pkg/cloud/libvirt/client/domain.go

```diff
@@ -105,6 +69,53 @@ func newDomainDef() libvirtxml.Domain {
 	return domainDef
 }
 
+func newDevicesDef() *libvirtxml.DomainDeviceList {
+	var serialPort uint
+
+	domainList := libvirtxml.DomainDeviceList{
+		Channels: []libvirtxml.DomainChannel{
+			{
+				Target: &libvirtxml.DomainChannelTarget{
+					VirtIO: &libvirtxml.DomainChannelTargetVirtIO{
+						Name: "org.qemu.guest_agent.0",
+					},
+				},
+			},
+		},
+		RNGs: []libvirtxml.DomainRNG{
+			{
+				Model: "virtio",
+				Backend: &libvirtxml.DomainRNGBackend{
+					Random: &libvirtxml.DomainRNGBackendRandom{},
+				},
+			},
+		},
+		Consoles: []libvirtxml.DomainConsole{
+			{
+				Source: &libvirtxml.DomainChardevSource{
+					Pty: &libvirtxml.DomainChardevSourcePty{},
+				},
+				Target: &libvirtxml.DomainConsoleTarget{
+					Type: "virtio",
+					Port: &serialPort,
+				},
+			},
+		},
+	}
+
+	if runtime.GOARCH != "s390x" {
```

Comment:
Why does S390 need a graphic and not others?

Reply:
Graphics/Spice isn't supported on s390x, so I just exclude it from s390x platform.

Reply:
ah, i misread the condition. :)